crates/topos-config/src/node.rs

```diff
@@ -26,31 +28,97 @@ pub struct NodeConfig {
     pub tce: Option<TceConfig>,
     pub sequencer: Option<SequencerConfig>,
     pub edge: Option<EdgeConfig>,
+
+    #[serde(skip)]
+    pub home_path: PathBuf,
+
+    #[serde(skip)]
+    pub node_path: PathBuf,
+
+    #[serde(skip)]
+    pub edge_path: PathBuf,
 }
 
 impl NodeConfig {
-    pub fn new<S: Serialize>(home: &Path, config: Option<S>) -> Self {
-        let base = load_config::<BaseConfig, _>(home, config);
+    /// Try to create a new node config struct from the given home path and node name.
+    /// It expects a config file to be present in the node's folder.
+    ///
+    /// This `config.toml` can be generated using: `topos node init` command
+    pub fn new<S: Serialize>(
+        home_path: &Path,
+        name: &str,
+        config: Option<S>,
+    ) -> Result<Self, std::io::Error> {
+        let node_path = home_path.join("node").join(name);
+        let config_path = node_path.join("config.toml");
+
+        // TODO: Move this to `topos-node` when migrated
+        if !Path::new(&config_path).exists() {
+            println!(
+                "Please run 'topos node init --name {name}' to create a config file first for \
+                 {name}."
+            );
+            std::process::exit(1);
+        }
+
+        Self::build_config(node_path, home_path, config)
+    }
+
+    /// Try to create a new node config struct from the given home path and node name.
+    ///
+    /// It doesn't check the existence of the config file.
+    /// It's useful for creating a config file for a new node, relying on the default values.
+    pub fn new_unchecked<S: Serialize>(
+        home_path: &Path,
+        name: &str,
```

Comment:
```suggestion
        node_folder: &str,
```

Reply:
I will rename to `node_name` as it is the expected parameter, if it was a `PathBuf` `node_folder` would work